cmd/sealer/cmd/cluster/run.go

```diff
@@ -180,12 +192,23 @@ func runWithClusterfile(clusterFile string, runFlags *types.RunFlags) error {
 	}
 
 	return kubeInstaller.Install(imageName, KubeInstallOptions{
-		RunMode:     runFlags.Mode,
-		IgnoreCache: runFlags.IgnoreCache,
+		RunMode:         runFlags.Mode,
+		IgnoreCache:     runFlags.IgnoreCache,
+		P2PDistribution: p2p,
 	})
 }
 
 func runWithArgs(imageName string, runFlags *types.RunFlags) error {
+	var p2p bool
+	switch runFlags.Distributor {
```

Comment:
When we start writing the same code a second time, we can consider whether we should extract it as a public method

Reply:
Yeah that's perfectly reasonable. However this part will be rewritten soon with the introduction of the other distribution methods I mentioned yesterday, I'd keep it simple here. The p2p variable itself will be gone then. There will be a set of logics which will be abstracted as a public method.